### ISSUE TITLE
Touch each file only once per action

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#124]
### TL;DR

As a template developer, I want to avoid accidentally double-processing any files in a single action.

Currently, if a file appears multiple times in in the `paths` list for an action, it will be re-processed. This is probably not what the template author intended. In the following example, if `first_directory` and `second_directory` refer to the same directory, then the files in that directory will be un-intentionally subjected to repeated `string_replace`ing.

```
action: 'string_replace'
params:
  paths: ['{{.first_directory}}', '{{.second_directory}}']
  replacements:
    // snip
```

The desired behavior would be to touch each file only once per `action`. We can keep a tracker (a map maybe) of which files (or directories?) have already been visited for the current action.

Discussed offline with @pdewilde . We don't think a template developer would ever want the opposite behavior where a file could be double-processed.
